crates/core/src/subscription/subscription.rs

```diff
@@ -101,6 +102,37 @@ impl SupportedQuery {
     pub fn as_expr(&self) -> &QueryExpr {
         self.as_ref()
     }
+
+    pub fn return_table(&self) -> TableId {
+        self.expr.source.get_db_table().unwrap().table_id
+    }
+
+    pub fn return_name(&self) -> String {
+        self.expr.source.table_name().to_owned()
+    }
+
+    pub fn filter_table(&self) -> TableId {
```

Comment:
What is the purpose of this filtering?

Reply:
I'm maintaining a map from TableIds to the queries that read from them. This method is really only relevant for joins where there is a return table and then this filter table, which is the joining table. I will add a clarifying comment here.